resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/session/TestBug297635.java

```diff
@@ -58,86 +104,73 @@ public void test1() {
 		} catch (IOException e) {
 			fail("1.2", e);
 		}
+	}
 
-		// register a save participant for the bundle
+	private void addSaveParticipant(String failMessage) {
 		try {
 			getWorkspace().addSaveParticipant(BUNDLE01_ID, TestBug297635.this);
 		} catch (CoreException e) {
-			fail("2.0", e);
+			fail(failMessage, e);
 		}
+	}
 
-		// create a project with a file
-		IProject project = getWorkspace().getRoot().getProject("Project1");
-		IFile file = project.getFile("file1.txt");
-		ensureExistsInWorkspace(project, true);
-		ensureExistsInWorkspace(file, getRandomContents());
-
-		// perform a full save
+	private void saveUsingWorkspace() {
 		try {
 			getWorkspace().save(true, getMonitor());
 		} catch (CoreException e) {
 			fail("3.0", e);
 		}
 	}
 
-	@SuppressWarnings("unchecked")
-	public void test2() {
+	private void reinstallBundle() {
 		// install the bundle again
-		// we need to restart the org.eclipse.core.resources bundle to read the tree file again
-		// we rely on the fact that the core.resources bundle doesn't save the tree when it is stopped
+		// we need to restart the org.eclipse.core.resources bundle to read the tree
+		// file again
+		// we rely on the fact that the core.resources bundle doesn't save the tree when
+		// it is stopped
 		try {
 			Bundle coreResourcesBundle = Platform.getBundle(ResourcesPlugin.PI_RESOURCES);
 			coreResourcesBundle.stop(Bundle.STOP_TRANSIENT);
-			Bundle b = BundleTestingHelper.installBundle("1", getContext(), ContentTypeTest.TEST_FILES_ROOT + "content/bundle01");
-			BundleTestingHelper.resolveBundles(getContext(), new Bundle[] {b});
+			Bundle b = BundleTestingHelper.installBundle("1", getContext(),
+					ContentTypeTest.TEST_FILES_ROOT + "content/bundle01");
+			BundleTestingHelper.resolveBundles(getContext(), new Bundle[] { b });
 			coreResourcesBundle.start(Bundle.START_TRANSIENT);
 		} catch (MalformedURLException e2) {
-			fail("1.0", e2);
```

Comment:
Maybe have a followup comming which uses better naming for the error message, for example the exception class as String? I know this is used a lot in the test but I never understood the value of using increasing numbers as error messages.

Reply:
The question is more why catch the exception at all and not simply throw it ?!

Reply:
I will, thank you.
I'm also not sure why the numbering but I noticed it's even part of the code templates (if I choose to handle an exception with "surround with try-catch" in a test then the `fail(<number>, e)` is added automatically). Maybe some attempt to quickly identify the line where the exception happened if no stack-traces were present? ... Anyhow, good to know I don't need to continue the tradition :)

Reply:
> Maybe some attempt to quickly identify the line where the exception happened if no stack-traces were present?

Note, the code was written probably at Java 1.2 times, may be ~20 years ago. Whatever didn't worked at that time in Java / Eclipse we will not know anymore, but today such style is simply not needed and wrong.

> it's even part of the code templates

Please create a dedicated PR to change templates to use defaults.

Reply:
@iloveeclipse , I updated the templates in #466